Varela Alan Christian Emmanuel
# Mismo programa pero en C, hecho con la IA de Google

In [ ]:
%%c
#include <stdio.h>

long long factorial(int n) {
    long long res = 1;
    for (int i = 2; i <= n; i++)
        res *= i;
    return res;
}

int main() {
    int num = 5;
    long long result = factorial(num);
    printf("Factorial = %lld\n", result);
    return 0;
}

UsageError: Cell magic `%%c` not found.


# A continuación se muestran los pasos necesarios para que el código funcione ejecutados de manera automática:

# Task
Compile and run the C code provided in the file "hello_world.c".

## Install necessary tools

### Subtask:
Install a C compiler and potentially other tools needed to compile and run C code.


**Reasoning**:
Install the build-essential package using apt-get to get a C compiler and other necessary tools.



In [ ]:
get_ipython().system('apt-get update')
get_ipython().system('apt-get install build-essential -y')

Hit:1 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease
Hit:2 https://cli.github.com/packages stable InRelease
Get:3 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease [3,632 B]
Get:4 http://security.ubuntu.com/ubuntu jammy-security InRelease [129 kB]
Hit:5 http://archive.ubuntu.com/ubuntu jammy InRelease
Get:6 https://r2u.stat.illinois.edu/ubuntu jammy InRelease [6,555 B]
Get:7 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [128 kB]
Hit:8 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease
Hit:9 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy InRelease
Hit:10 https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy InRelease
Get:11 http://security.ubuntu.com/ubuntu jammy-security/universe amd64 Packages [1,287 kB]
Get:12 https://r2u.stat.illinois.edu/ubuntu jammy/main all Packages [9,373 kB]
Get:13 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [127 kB]
Get:14 http://ar

## Write the c code

### Subtask:
Provide the C code in a file.


**Reasoning**:
Write the C code to a file named "hello_world.c".



In [ ]:
with open('hello_world.c', 'w') as f:
    f.write("""
#include <stdio.h>

long long factorial(int n) {
    long long res = 1;
    for (int i = 2; i <= n; i++)
        res *= i;
    return res;
}

int main() {
    int num = 5;
    long long result = factorial(num);
    printf("Factorial = %lld\\n", result);
    return 0;
}
""")

## Compile the c code

### Subtask:
Compile the C code using the installed compiler.


**Reasoning**:
Compile the C code file `hello_world.c` using gcc and name the output executable `hello_world`.



In [ ]:
get_ipython().system('gcc hello_world.c -o hello_world')

## Run the compiled code

### Subtask:
Execute the compiled C program.


**Reasoning**:
Execute the compiled C program using the system command.



In [ ]:
get_ipython().system('./hello_world')

Factorial = 120


## Summary:

### Data Analysis Key Findings

*   The necessary build tools, including a C compiler, were already installed on the system.
*   A C program to calculate the factorial of 5 and print the result was successfully written to the file `hello_world.c`.
*   The C code in `hello_world.c` was successfully compiled using `gcc`, creating an executable file named `hello_world`.
*   Executing the compiled program `hello_world` produced the output "Factorial = 120".

### Insights or Next Steps

*   The task of compiling and running the provided C code was successfully completed.
*   The program correctly calculated the factorial of 5 as 120.
